src/main/kotlin/network/warzone/mars/utils/Gson.kt

```diff
@@ -1,21 +1,69 @@
 package network.warzone.mars.utils
 
-import com.google.gson.Gson
-import com.google.gson.GsonBuilder
-import com.google.gson.TypeAdapter
+import com.google.gson.*
 import com.google.gson.reflect.TypeToken
 import com.google.gson.stream.JsonReader
 import com.google.gson.stream.JsonWriter
 import com.tinder.scarlet.Message
 import com.tinder.scarlet.MessageAdapter
+import network.warzone.api.database.models.AgentParams
+import network.warzone.mars.api.socket.models.PlayerUpdateData
+import network.warzone.mars.match.tracker.PlayerBlocks
+import network.warzone.mars.player.achievements.models.AchievementParent
 import okio.Buffer
+import org.bukkit.Material
 import java.io.OutputStreamWriter
 import java.io.StringReader
 import java.lang.reflect.Type
 import java.nio.charset.StandardCharsets.UTF_8
+import java.util.*
 
-val GSON = GsonBuilder()
-    .create()!!
+val GSON_CFG : GsonBuilder.() -> Unit = {
+    registerTypeAdapter(Date::class.java, JsonDeserializer { json, _, _ ->
+        Date(json.asJsonPrimitive.asLong)
+    })
+    registerTypeAdapter(Date::class.java, JsonSerializer<Date> { date, _, _ ->
+        JsonPrimitive(date.time)
+    })
+    registerTypeAdapter(
+        AgentParams::class.java,
+        ClosedPolymorphismDeserializer.createFromSealedClass(AgentParams::class)
+    )
+    registerTypeAdapter(
+        PlayerUpdateData::class.java,
+        ClosedPolymorphismDeserializer.createFromSealedClass(PlayerUpdateData::class)
+    )
+    registerTypeAdapter(
+        AchievementParent::class.java,
+        ClosedPolymorphismDeserializer.createFromSealedClass(AchievementParent::class)
+    )
+    registerTypeAdapter(
+        PlayerBlocks::class.java,
```

Comment:
Fine for now but in the future should not be necessary if we have a consistent definition for this model